### PR TITLE
Cscfairmeta 463 and Cscfairmeta 462 add provenance and related resource

### PR DESCRIPTION
**Pull request by ismotee**
Add provenance and add related resource in four steps:
- expand/collapse logic for fields (CSCFAIRMETA-573)
- multiple modal system (CSCFAIRMETA-574)
- add provenance modal (CSCFAIRMETA-575)
- backend post and patch update for provenance (CSCFAIRMETA-576)

expand/collapse logic for fields added

### CODE REVIEW

File: etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import ActorsList from './actorsList'
+import { Actor } from '../../../../../stores/view/qvain.actors'
+import { ROLE } from '../../../../../utils/constants'
+
+const ActorsInput = ({ Stores, language }) => {
+  const { Actors, Provenances } = Stores.Qvain
+
+  const CreateOption = { Option: CustomOption }
+
+  const options = [{ value: 'create-actor', label: 'Luo uusi toimija' }, ...Actors.actorOptions.map(
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx]: added translate()

File: etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            />
+          </ButtonContainer>
+        </>
+        )}
+    </Observer>
+    )
+}
+
+const CustomOption = ({ children, ...props }) => (
+  <>
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx]: removed fragment tags

File: etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsList.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        )}
+        </ButtonContainer>
+      </ButtonGroup>
+    )
+  })
+
+  return (
+    <>
+      {items.length === 0 &&
+        (null)
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsList.jsx]: removed size check and changed to return actorItems only

File: etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    setSelectedActor(selection)
+  }
+
+  const setSelectedActor = value => {
+    Stores.Qvain.Provenances.changeAttribute('selectedActor', value)
+  }
+
+  return (
+    <Observer>
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx]: removed <Observer />

File: etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+CustomOption.propTypes = {
+  children: PropTypes.node.isRequired,
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number
+  ]).isRequired
+}
+
+const EditIcon = styled(FontAwesomeIcon).attrs(() => ({
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/history/provenance/form/actorsInput.jsx]: removed `() =>`

File: etsin_finder/frontend/js/components/qvain/submitButtons.jsx [hunk 6]
```diff
@@ -208,6 +208,15 @@ class SubmitButtons extends Component {
     return values
   }
 
+
+  submit = async (submitFunction) => {
+    const { Stores } = this.props
+    const isProvenanceActorsOk = await Stores.Qvain.checkProvenanceActors()
+    console.log('ok? ', isProvenanceActorsOk)
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/submitButtons.jsx]: removed `console.log`

File: etsin_finder/frontend/js/components/qvain/submitButtons.jsx [hunk 7]
```diff
@@ -589,14 +597,14 @@ class SubmitButtons extends Component {
     if (original && original.state !== 'draft') {
       // published -> draft
       submitDraft = (
-        <SubmitButton disabled={disabled} onClick={() => this.handleSaveAsDraft()}>
+        <SubmitButton disabled={disabled} onClick={() => this.submit(this.handleSaveAsDraft())}>
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/submitButtons.jsx]: removed function call

File: etsin_finder/frontend/js/components/qvain/submitButtons.jsx [hunk 8]
```diff
@@ -589,14 +597,14 @@ class SubmitButtons extends Component {
...
           <Translate content="qvain.saveDraft" />
         </SubmitButton>
       )
 
       // published -> published
       submitPublished = (
-        <SubmitButton disabled={disabled} onClick={() => this.handleUpdate()}>
+        <SubmitButton disabled={disabled} onClick={() => this.submit(this.handleUpdate())}>
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/submitButtons.jsx]: removed function call

File: etsin_finder/frontend/js/components/qvain/temporalAndSpatial/spatial/handleSave.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+export default async (Field, translationsRoot) => {
+  const { inEdit, save, clearInEdit, setValidationError } = Field
+  const altNanError = `${translationsRoot}.error.altitudeNan`
+  try {
+    await spatialNameSchema.validate(inEdit.name)
+  } catch (e) {
+    setValidationError(e.message)
+    return;
+  }
+  // number validation translation cannot be placed into validationForm because number() doesn't have message argument
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/temporalAndSpatial/spatial/handleSave.js]: Added `typeError` and removed explicit altitudeNan try-catch

File: etsin_finder/frontend/locale/english.js [hunk 10]
```diff
@@ -1030,24 +1054,233 @@ const english = {
...
+            },
+            en: {
+              label: 'Outcome description',
+              placeholder: 'Outcome description (in English)',
+            },
+          },
+          periodOfTimeInput: {
+            label: 'Period of time',
+            startPlaceholder: 'starts',
+            endPlaceholder: 'ends',
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/locale/english.js]: fixed as suggested

**Review by tahme — DISMISSED** [review]
Seems to work ok now :+1:

**Review by tahme — APPROVED** [review]

**Pull request merged**